### PR TITLE
New Mexico Property Tax Rebate for persons 65 or older

### LINKED ISSUE #2427: New Mexico Property Tax Rebate
[Tax Form Instructions](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf) page 58
[Legislative Reference](https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/856ebf4b-3814-49dd-8631-ebe579d6a42b/Personal%20Income%20Tax.pdf) page 57

### PR DESCRIPTION
**Pull request by XZ-mabel**
Fixes #2427
Created nm folder

### Summary
📝📱🐛

Add a new policy parameter `child_allowance` to the US tax and benefit system. Update the `household_net_income` function to include the child allowance in the calculation. Add tests and documentation for the new parameter and function.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `ubi_amount` to the UBI reform function ()

### CODE REVIEW

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+description: New Mexico minimum age eligible for property tax rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:1]:
```suggestion
description: New Mexico provides the property tax rebate for filer at or above this age threshold.
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2022-01-01: 65
+metadata:
+  unit: year
+  period: year
+  label: NM minimum age for property tax rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:7]:
```suggestion
  label: New Mexico minimum age for property tax rebate 
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+description: New Mexico minimum age eligible for property tax rebate
+values:
+  2022-01-01: 65
+metadata:
+  unit: year
+  period: year
+  label: NM minimum age for property tax rebate 
+  reference:
+    - title: NM 2022 PIT-RC instructions, page 2
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:10]: please add [legal code ](https://casetext.com/regulation/new-mexico-administrative-code/title-3-taxation/chapter-3-personal-income-taxes)reference

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:10]: all yaml files

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/marriage_status.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  period: year
+  breakdown:
+  label: NM maximum property tax rebate amount
+  reference: 
+    - title: NM 2022 PIT-RC instructions, page 2
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf
+Unmarried: 
+  2022-01-01: 250
+Married:
+  2022-01-01: 125
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/marriage_status.yaml:13]: add 2021 values + tax form reference, all yaml files

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/marriage_status.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+description:  New Mexico maximum values for married or unmarried person 
+metadata:
+  unit: currency-USD
+  period: year
+  breakdown:
+  label: NM maximum property tax rebate amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/marriage_status.yaml:6]:
```suggestion
  label: New Mexico maximum property tax rebate amount
```

File: policyengine_us/parameters/gov/states/la/tax/credits/nonrefundable.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/credits/nonrefundable.yaml:1]: delete

File: policyengine_us/parameters/gov/states/la/tax/credits/refundable.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/credits/refundable.yaml:1]: delete

File: policyengine_us/variables/gov/states/la/tax/credits/la_nonrefundable.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/credits/la_nonrefundable.py:1]: delete

File: policyengine_us/variables/gov/states/la/tax/credits/la_refundable.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/credits/la_refundable.py:1]: delete

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/maximum_modified_gross_income.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+description: New Mexico maximum modified gross income for property tax rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/maximum_modified_gross_income.yaml:1]:
```suggestion
description: New Mexico provides the property tax rebate for filers with income below this threshold.
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/maximum_modified_gross_income.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2022-01-01: 1_600
+metadata:
+  unit: currency-USD
+  period: year
+  label: NM maximum mgi for property tax rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/maximum_modified_gross_income.yaml:7]:
```suggestion
  label: New Mexico property tax rebate max income
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/rate.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+description: New Mexico property tax rebate rate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/rate.yaml:1]:
```suggestion
description: New Mexico provides this percentage of rent paid as the property tax rebate.
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/rate.yaml [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+values:
+  2022-01-01: 0.06
+metadata:
+  unit: /1
+  period: year
+  label: NM property tax rebate rate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/rate.yaml:7]:
```suggestion
  label: New Mexico property tax rebate rate
```

File: policyengine_us/variables/gov/states/nm/tax/income/rebates/nm_property_tax_rebate.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class nm_property_tax_rebate(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "New Mexico property tax rebate"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf"
+    default_value = 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nm/tax/income/rebates/nm_property_tax_rebate.py:11]:
```suggestion
    defined_for = StateCode.M
```

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  unit: year
+  period: year
+  label: New Mexico minimum age for property tax rebate 
+  reference:
+    - title: NM 2022 PIT-RC instructions, page 2
+      href: https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf
+    - title: 2021 PIT-RC NEW MEXICO REBATE AND CREDIT SCHEDULE
+      href: https://taxsim.nber.org/historical_state_tax_forms/NM/2021/2021pit-rc.pdf
+    - legal_code: R. § 3.3.13.8
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:13]: need to add full title and href - same as lines 11 and 12

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:13]: all yaml files

File: policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml [hunk 12]
```diff
@@ -0,0 +1,14 @@
+description: New Mexico provides the property tax rebate for filer at or above this age threshold.
+values:
+  2021-01-01: 65
+metadata:
+  unit: year
+  period: year
+  label: New Mexico minimum age for property tax rebate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nm/tax/income/rebates/property_tax/age.yaml:7]:
```suggestion
  label: New Mexico property tax rebate age threshold
```

File: policyengine_us/variables/gov/states/nm/tax/income/rebates/nm_property_tax_rebate_eligible.py [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    definition_period = YEAR
+    reference = "https://klvg4oyd4j.execute-api.us-west-2.amazonaws.com/prod/PublicFiles/34821a9573ca43e7b06dfad20f5183fd/1afc56af-ea90-4d48-82e5-1f9aeb43255a/PITbook2022.pdf"
+    defined_for = StateCode.NM
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.nm.tax.income.rebates.property_tax
+        # Head or spoue eligible if 65 or over
+        age_head = tax_unit("age_head", period)
+        age_spouse = tax_unit("age_spouse", period)
+        age_eligible = (age_head | age_spouse) >= p.age_eligibility
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/nm/tax/income/rebates/nm_property_tax_rebate_eligible.py:17]: I'm surprised the test passes as-is.
```suggestion
        head_age_eligible = age_head >= p.age_eligibility
        spouse_age_eligible = age_spouse >= p.age_eligibility
        age_eligible = head_age_eligible | spouse_age_eligible
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**